stl/src/mutex.cpp

```diff
@@ -32,11 +32,9 @@ extern "C" _CRTIMP2_PURE void _Thrd_abort(const char* msg) { // abort on precond
 #define _THREAD_ASSERT(expr, msg) ((void) 0)
 #endif // _THREAD_CHECKX
 
-__stl_sync_api_modes_enum __stl_sync_api_impl_mode = __stl_sync_api_modes_enum::normal;
-
-extern "C" _CRTIMP2 void __cdecl __set_stl_sync_api_mode(__stl_sync_api_modes_enum mode) {
-    __stl_sync_api_impl_mode = mode;
-}
+// TRANSITION, ABI: preserved for binary compatibility
+enum class __stl_sync_api_modes_enum { normal, win7, vista, concrt };
+extern "C" _CRTIMP2 void __cdecl __set_stl_sync_api_mode(__stl_sync_api_modes_enum) {}

```

Comment:
We must preserve `__set_stl_sync_api_mode()` because it's exported (controlled by `_CRTIMP2`):

```
C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.33.31424\bin\Hostx64\x64>dumpbin /exports msvcp140.dll | grep -P __set_stl_sync_api_mode
       1514  5E9 000136B0 __set_stl_sync_api_mode
```

We can remove the global variable `__stl_sync_api_impl_mode` because it was never declared in user-visible headers (i.e. `stl/inc` didn't mention it). It was used within `stl/src` only, so there are no mix-and-match concerns (there's only one STL DLL and static LIB, we don't have to worry about old-and-new).